### PR TITLE
Add GitHub Actions CI support.

### PR DESCRIPTION
**Pull request by XhmikosR**
1. this requires #61 for the lock file addition. If you guys don't want a lockfile, I can change this to just use `npm i`
2. I don't know what exactly is required for tests to run. A token?
3. ~I also plan to add Windows testing later~ -- DONE, but I might need to drop it from here and tackle it later separately

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,33 +6,43 @@
...
     "lint": "standard",
-    "test": "standard && tape test.js"
+    "unit": "tape test.js",
+    "test": "npm run lint && npm run unit"
   },
-  "devDependencies": {
-    "standard": "~12.0.1",
-    "tape": "~4.9.1"
-  }
+  "files": [
```
Comment (rvagg) [inline package.json]: This is the only thing I'm not good with in this PR; you're excluding readme, license and tests in this, to only save 1071 bytes in the packed tarball. We should be shipping an entire, stand-alone package.

Comment (XhmikosR) [reply package.json]: First of all this change you are talking about is on #61 :P

Secondly, no, I don't exclude those. npm includes some stuff by default. I think you are confusing something there.

```
npm notice === Tarball Contents ===
npm notice 4.1kB changelog-maker.js
npm notice 1.9kB collect-commit-labels.js
npm notice 2.4kB commit-to-output.js
npm notice 488B  group-commits.js
npm notice 1.7kB groups.js
npm notice 371B  reverts.js
npm notice 1.2kB package.json
npm notice 1.1kB LICENSE.md
npm notice 4.7kB README.md
```

Comment (rvagg) [reply package.json]: so you're just excluding test.js, -1 from me, don't exclude anything, this project doesn't have anything of onerous size and is not intended to be a dependency for anything else

**Comment by XhmikosR** [comment]
Alright, let's focus on getting this up and running.

I untied this patch from #61. I have added Ubuntu only and I'm using `npm install` for starters.

**Comment by XhmikosR** [comment]
If someone could add the needed secrets on the repo secrets page, let me know what names you used and I'll adapt the patch so that it uses them when running tests.

**Comment by rvagg** [comment]
according to https://help.github.com/en/articles/virtual-environments-for-github-actions#github_token-secret we _might_ be able to make use of `GITHUB_TOKEN`. You'll need to do this in the action:

```
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Then if you can run some bash before running the tests, try this (for Linux):

```
mkdir -p .config/changelog-maker/ && echo '{"user":"nodejs","token":"'${GITHUB_TOKEN}'"}' > .config/changelog-maker/config.json
```

The awkward bit is the `"user"`, I'm not sure whether it matters. It gets combined with the token for the `Authorization: Basic <base64 of user:token>` header (see https://github.com/rvagg/ghutils/blob/master/ghutils.js#L13). I don't see what user `GITHUB_TOKEN` is associated with or even whether it matters.

**Comment by XhmikosR** [comment]
Alright, `GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"` doesn't seem to work.

We need to add and pass the exact variables including the user.

**Comment by rvagg** [comment]
if that doesn't work, the alternatives i see are: (1) generate a token for an account owned by the org or (2) modify ghutils to directly use the `Authorization: token <token>` header. I'm unclear about whether this will break existing use of ghutils but we could probably find out.

**Comment by rvagg** [comment]
@XhmikosR you need to do something with `GITHUB_TOKEN`, see the bash I put in that previous comment to put it into a config file that's picked up by [ghauth](https://ghub.io/ghauth).

**Comment by rvagg** [comment]
```
       - run: mkdir -p .config/changelog-maker/ && echo '{"user":"nodejs","token":"'${GITHUB_TOKEN}'"}' > .config/changelog-maker/config.json
       - run: npm test
```

?

**Comment by XhmikosR** [comment]
@rvagg does ghauth pick up user variables about user and token or does it need a config explicitly?

I will try this later, but I'm on Windows currently.

**Comment by rvagg** [comment]
@XhmikosR it picks it up from a config file that looks like `{"user":"foo","token":"bar"}`. It should be created on Windows @ `%LOCALAPPDATA%/changelog-maker/config.json` (as per https://github.com/LinusU/node-application-config#config-location). When you first invoke `ghauth` (see collect-commit-labels.js) and you don't have this file, it should prompt you on the commandline to enter github credentials and it'll make a token for you and save it to that file for future use.

**Comment by XhmikosR** [comment]
@rvagg so as things are right now, we need tho create these configs. Would it be possible to make ghauth use env vars too? I think it would simplify things a lot.

Regardless, I'll experiment with this and let you know how it goes.

**Comment by rvagg** [comment]
> Would it be possible to make ghauth use env vars too

yes, it would certainly simplify the CI case .. I'm not sole maintainer so any such PR will need discussion.

**Comment by XhmikosR** [comment]
It works! The path is `~/.config/changelog-maker/config.json`.

**Comment by XhmikosR** [comment]
Only thing I don't know is if this will work for forks. I guess we'll find out later.

/CC @targos for a quick review too :)

**Comment by XhmikosR** [comment]
I added Windows too. I don't think the shell used on Windows demonstrates the failures I had locally, though, because CI is passing even with the old gitexec.

That being said, this should better than nothing.

**Pull request merged**